src/Bicep.Core/TypeSystem/DiscriminatedObjectType.cs

```diff
@@ -32,12 +37,12 @@ public DiscriminatedObjectType(string name, string discriminatorKey, IEnumerable
 
             this.UnionMembersByKey = unionMembersByKey.ToImmutableDictionary();
             this.DiscriminatorKey = discriminatorKey;
-            this.DiscriminatorKeysUnionType = UnionType.Create(unionKeyTypes);
+            this.DiscriminatorKeysUnionType = UnionType.Create(unionKeyTypes.Select(x => x));
```

Comment:
>.Select(x => x) [](start = 76, length = 15)

Does `x => x` do anything?

Reply:
I have no recollection of doing this! But I assume I did this to match one of the IEnumerable overloads for `UnionType.Create`. Let me see if I can find a better solution.

Reply:
If it doesn't compile after removing the Select(), I'd leave as-is.

---
In reply to: [492361799](https://github.com/Azure/bicep/pull/536#discussion_r492361799) [](ancestors = 492361799)